### PR TITLE
Extend v2 pipeline to test different OS versions

### PR DESCRIPTION
**Pull request by rukawata**
To be able to run the test cases for different OS versions, below changes are made in this PR:
- Separate jobs for different scenarios as template under `template/job-template-per-os.yaml`
- Use the above template trigger test jobs for two OS versions - can be extended in the future:
   - **SLES12**
   - **RHEL7**
  The above two versions are currently supported by the Ansible [PR](https://github.com/Azure/sap-hana/pull/250).
- Adjust the cleanUp job so it can work with the new naming conversion for created resource group: `"${scenario}-${os}-$(Build.BuildId)"` eg. `v2-reuseNSG-RHEL7-1365`
- Adjust the `README.MD` accordingly.

### CODE REVIEW

File: templates/job-template-per-os.yaml [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+  - script: |
+      az login --service-principal --user $(hana-pipeline-spn-id) --password  $(hana-pipeline-spn-pw) --tenant $(landscape-tenant) --output none
+      echo '##vso[task.setvariable variable=arm_id]$(az network nsg show --resource-group v2-allNew-${{parameters.osVersion}}-$(Build.BuildId) --name nsg-mgmt --query id --output tsv)' 
+  - template: update-place-holder.yaml
+    parameters:
+      scenario: $(scenario)
+      testCaseName: $(testcase)
+      placeHolder: var-rg-name
+      value: $(testcase)
+  - template: update-place-holder.yaml
```
Comment (persiaAziz) [inline templates/job-template-per-os.yaml]: Can we rename the update-place-holder.yaml to something else in this PR. Specifically get rid of the "place-holder" and name it something like change resource change param or something?

Comment (rukawata) [reply templates/job-template-per-os.yaml]: Sure! I have some options, let me know which causes less confusion :)

- update-template-json
- update-template-value
- replace-json-value
- ...

File: azure-pipelines-v2.yaml [hunk 2]
```diff
@@ -150,26 +64,37 @@ stages:
 # Tag "Delete=True" will be added to mark all resource groups for   #
 # scheduled deletion done by azure-pipelines-cleanup.yaml           #
 #####################################################################
-- stage: DestroyingResources
+- stage: DestoryingResources
```
Comment (pabowers) [inline azure-pipelines-v2.yaml]: typo: should be `DestroyingResources`

File: templates/job-template-per-os.yaml (hunk 1 above)
Comment (rukawata) [reply templates/job-template-per-os.yaml]: Change it to test-case-template-update.

File: azure-pipelines-v2.yaml (hunk 2 above)
Comment (rukawata) [reply azure-pipelines-v2.yaml]: Thanks! Adjusted.

**Pull request merged**